code/code/disc/disc_wrath.cc

```diff
@@ -15,103 +15,102 @@
-    swarm = Mob::LOCUSTS40;
+    vnumToLoad = Mob::LOCUSTS40;
   else 
-    swarm = Mob::LOCUSTS50;
+    vnumToLoad = Mob::LOCUSTS50;
 
-  if (!(locusts = read_mobile(swarm, VIRTUAL))) {
+  TMonster *locusts = read_mobile(vnumToLoad, VIRTUAL);
+
+  if (!locusts) {
     vlogf(LOG_BUG, "Spell PLAGUE_LOCUSTS unable to load mob...");
     caster->sendTo("Unable to create the locusts, please report this.\n\r");
     act("Nothing seems to happen.", FALSE, caster, NULL, NULL, TO_ROOM);
     return SPELL_FAIL;
-  }
-
-  if ((caster->followers) && (level < GOD_LEVEL1))  {
-    for (k = caster->followers; k; k = n) {
-      n = k->next;
-      if (isname("locust", k->follower->name)) {
-        act("Nothing seems to happen.", FALSE, caster, NULL, NULL, TO_ROOM);
-        caster->sendTo("You would have to be a god to summon another locust swarm!\n\r");
-        delete locusts;
-        locusts=NULL;
-        return SPELL_FAIL;
-      }
-    }
-  }
+  }  
 
-  level = min(level, 75);
+  float normalizedLevel = max(1.0, min(level, 75) / 2.0);
 
-  float lvl = max(1.0, level/2.0);
   locusts->setDamPrecision(20);
-
   locusts->setExp(0);
   locusts->setMoney(0);
   locusts->setHitroll(0);
 
   *caster->roomp += *locusts;
 
-  act("You hear a loud humming sound as a swarm of locusts decends from above.", TRUE, locusts, NULL, caster, TO_ROOM);
-  act("You hear a loud humming sound as a swarm of locusts decends from above.", TRUE, locusts, NULL, caster, TO_CHAR);
+  act("You hear a loud humming sound as a swarm of locusts decends from above!", TRUE, locusts, NULL, caster, TO_ROOM);
+  act("You hear a loud humming sound as a swarm of locusts decends from above!", TRUE, locusts, NULL, caster, TO_CHAR);
 
   if (caster->bSuccess(bKnown, caster->getPerc(), SPELL_PLAGUE_LOCUSTS)) {
     switch(critSuccess(caster, SPELL_PLAGUE_LOCUSTS)) {
       case CRIT_S_KILL:
       case CRIT_S_TRIPLE:
       case CRIT_S_DOUBLE:
         CS(SPELL_PLAGUE_LOCUSTS);
-        act("The swarm seems to be especially angry and to be circling especially fast!", TRUE, caster, 0, locusts, TO_ROOM);
-        caster->sendTo("The swarm seems to be especially angry and to be circling especially fast!\n\r");
-        lvl *= 2;
+        act("The swarm seems larger and angrier than expected!", TRUE, caster, 0, locusts, TO_ROOM);
+        caster->sendTo("The swarm seems larger and angrier than expected!\n\r");
+        normalizedLevel *= 2;
         break;
       case CRIT_S_NONE:
         if (victim->isLucky(caster->spellLuckModifier(SPELL_PLAGUE_LOCUSTS))) {
           SV(SPELL_PLAGUE_LOCUSTS);
-          lvl /= 2;
+          act("Luckily for $N, only a small swarm responds to $n's prayer!", TRUE, caster, 0, victim, TO_NOTVICT);
+          act("Luckily for you, only a small swarm responds to $n's prayer!", TRUE, caster, 0, victim, TO_VICT);
+          act("Unfortunately, only a small swarm responds to your prayer.", TRUE, caster, 0, victim, TO_CHAR);
+          normalizedLevel /= 2;
```

Comment:
I know you are just preserving the logic in place but we calculate a level for the mob then double it if it's a critical and half it if it's not?

I'd honestly just remove this line. If the plague is too strong we can adjust later :)

Reply:
Sounds good, I've never been a fan of this sort of crit fail/success code myself.

Reply:
Fixed in https://github.com/sneezymud/sneezymud/pull/353/commits/1067a974875a7cb81742309e23a76e9ce279cd6b